pkg/dispatcher/config.go

```diff
@@ -0,0 +1,127 @@
+package dispatcher
+
+import (
+	"io/ioutil"
+	"regexp"
+
+	"github.com/pkg/errors"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	prowconfig "k8s.io/test-infra/prow/config"
+	"sigs.k8s.io/yaml"
+)
+
+// Config is the configuration file of this tools, which defines the cluster parameter for each Prow job, i.e., where it runs
+type Config struct {
+	// the cluster context name if no other condition matches
+	Default string `json:"default"`
+	// the cluster context name for non Kubernetes jobs
+	NonKubernetes string `json:"nonKubernetes"`
+	// Groups maps a group of jobs to a cluster
+	Groups JobGroups `json:"groups"`
+	// BuildFarm maps groups of jobs to a cloud provider, like GCP
+	BuildFarm map[string]JobGroups `json:"buildFarm,omitempty"`
+}
+
+// JobGroups maps a group of jobs to a cluster
+type JobGroups = map[string]Group
```

Comment:
Same comment regarding the typing here, I think we will not often add/remove clusters and using a string type makes this easier to read and makes it easier to debug if we somehow end up with a non-existent cluster

Reply:
So we keep the string here, right?

Reply:
No, use a typed string for the cluster here as well

Reply:
Done.